tests/acceptance/pageObjects/FilesPageElement/appSideBar.js

```diff
@@ -39,7 +39,7 @@ module.exports = {
       )
       return isVisible
     },
-    isSideBarOpenForResource: async function (resource, elementType = 'any', timeout = 500) {
+    isSideBarOpenForResource: async function (resource, elementType = 'any', timeout = 5000) {
```

Comment:
Nightwatch APIs have default `5000ms` timeout to wait and retry for element visibility.

Reply:
The function is used with both `true` and `false` as expected outcome. Having a long timeout would slow down all use cases where we expect `false` as outcome.

Reply:
It's a bit of a no-win situation. When we expect true, it is good to have the timeout longer. If the UI runs a bit slow, then it is better to wait an extra bit and have the test pass, rather than having a timing-related fail.

When doing "negative" tests, there is not a perfect timeout! The assertion is actually that the element/resource never appears, and never is a long time. If the timeout is too short, then we actually might not detect the result correctly - the thing might appear another 1-second later.

I think that I remember in core tests somewhere that we give the test code knowledge of the expected outcome (exists/not-exists). If the thing is expected to exist, then we make the timeout longer. If the thing is not expected to exist then we make just a "moderate" timeout, not wasting too much time waiting for it to "never" exist.

We could code it like that.

Reply:
We do the same in other places already of the web ui acceptance tests already. If you change it that the timeout depends on the expected outcome, that would be good 👍

Reply:
@saw-jan ^

Reply:
Yeah, it would be great to have variable timeouts depending on the expected outcome.
I will implement previously `500ms` for `false` case and `5000ms` for `true` case.

It is not true that the test will wait for `5s` every time. It depends on how fast the element gets visible. If it gets visible within a second then the test will move to the next step without waiting for full 5s.